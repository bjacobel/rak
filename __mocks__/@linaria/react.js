export const styled = jest.fn(strings => () => strings);
