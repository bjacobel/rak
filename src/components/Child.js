import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from '../stylesheets/child.css';
import { link } from '../stylesheets/link.css';

export default class Child extends Component {
  render() {
    const { match } = this.props;
    return (
      <div>
        <h3 className={styles.id}>{`received param: ${match.params.id}`}</h3>
        <Link className={link} to="/">
          Home
        </Link>
      </div>
    );
  }
}

Child.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};
