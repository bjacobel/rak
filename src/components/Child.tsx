import React from 'react';
import { Link, useParams } from 'react-router-dom';

import { id } from '../stylesheets/child.css';
import { link } from '../stylesheets/link.css';

export default () => {
  const params = useParams<{ id: string }>();
  return (
    <div>
      <h3 className={id}>{`received param: ${params.id}`}</h3>
      <Link className={link} to="/">
        Home
      </Link>
    </div>
  );
};
