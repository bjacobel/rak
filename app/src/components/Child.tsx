import { styled } from '@linaria/react';
import { useParams } from 'wouter';

import { center, openSans } from '../stylesheets/shared.css';
import { lightgrey } from '../stylesheets/colors.css';
import StyledLink from './StyledLink';

const Child = styled.h3`
  ${center}
  ${openSans}
  ${lightgrey}
  font-style: italic;
`;

export default () => {
  const params = useParams<{ id: string }>();
  return (
    <div>
      <Child>{`received param: ${params.id}`}</Child>
      <StyledLink to="/">Home</StyledLink>
    </div>
  );
};
