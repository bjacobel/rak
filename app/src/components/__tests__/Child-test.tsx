import { ReactNode } from 'react';
import { Switch, Route } from 'wouter';

import Child from '../Child';
import { render } from '../../testing/rtl';

describe('child component', () => {
  let child: ReactNode;

  beforeEach(() => {
    child = (
      <Switch>
        <Route component={Child} path="/child/:id" />;
      </Switch>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(child, { route: '/child/foo' });

    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the param it recieves in props', async () => {
    const { getByText } = render(child, { route: '/child/foo' });

    expect(getByText('received param: foo')).toBeInTheDocument();
  });

  it('has a link back to home', async () => {
    const { getByRole } = render(child, { route: '/child/foo' });
    const homeLink = getByRole('link');

    expect(new URL((homeLink as HTMLLinkElement).href).pathname).toEqual('/');
  });
});
