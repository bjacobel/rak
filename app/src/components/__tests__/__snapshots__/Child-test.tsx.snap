// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`child component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      received param: foo
    </h3>
    <a
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;
